python/unitytrainers/trainer_controller.py

```diff
@@ -107,7 +107,10 @@ def _process_graph(self):
                 elif not self.trainers[brain_name].parameters["use_recurrent"]:
                     nodes += [scope + x for x in ["action", "value_estimate", "action_probs"]]
                 else:
-                    nodes += [scope + x for x in ["action", "value_estimate", "action_probs", "recurrent_out"]]
+                    node_list = ["action", "value_estimate", "action_probs", "recurrent_out"]
+                    if not self.trainers[brain_name].is_continuous:
+                        node_list.append("prev_action")
```

Comment:
Is there a scenario in which the developer will want to use prev_action but not recurrent ?

Reply:
Is prev_action an output of the network or is it an input ? If it is an input, then you should not need to add it here...

Reply:
Possibly, if they are using state stacking.

Reply:
Gotcha. Will fix.